### PR TITLE
Update gitpod dev image

### PR DESCRIPTION
**Pull request by aledbf**
## Description

- Trigger build to get the new workspace-full image with go v1.18.3
- Update k8s binaries to v1.24.1
- Update cert-manager to v1.8.0
- Update google-cloud-sdk to 388.0.0
- Update prometheus to v2.36.0

## Release Notes
```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: dev/image/Dockerfile [hunk 1]
```diff
@@ -80,7 +80,7 @@ RUN cd /usr/bin && curl -fsSL https://github.com/c4milo/github-release/releases/
 ### Protobuf
 RUN set -ex \
     && tmpdir=$(mktemp -d) \
-    && curl -fsSL -o $tmpdir/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-linux-x86_64.zip \
+    && curl -fsSL -o $tmpdir/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v21.0/protoc-21.0-linux-x86_64.zip \
```
Comment (sagor999) [inline dev/image/Dockerfile:83]: there is version v21.1, maybe we should upgrade to that one right away?

**Review by sagor999 — APPROVED** [review]

File: dev/image/Dockerfile (hunk 1 above)
Comment (aledbf) [reply dev/image/Dockerfile]: fixed

**Review by jeanp413 — APPROVED** [review]

**Review by mads-hartmann — APPROVED** [review]
🛹

**Comment by jankeromnes** [comment]
Hi there! 👋 Is it possible that this broke the `main` build?

```
protoc-gen-js: program not found or is not executable
```

https://werft.gitpod-dev.com/job/gitpod-build-main.3499

**Pull request merged**